src/pages/api/reviewed-preprints.page.ts

```diff
@@ -0,0 +1,162 @@
+
+  return reviewedEvent ? `${reviewedEvent.date}T03:00:00Z` : undefined;
+};
+
+export const writeResponse = (res: NextApiResponse, contentType: string, statusCode: 200 | 400 | 404, message: BadRequestMessage | ReviewedPreprintListResponse | ReviewedPreprintItemResponse) : void => {
+  res
+    .writeHead(statusCode, {
+      'Content-Type': contentType,
+      'Cache-Control': statusCode === 200 ? 'max-age=300, public, stale-if-error=86400, stale-while-revalidate=300' : 'must-revalidate, no-cache, private',
+      Vary: ['Accept', 'Authorization'],
+    })
+    .end(JSON.stringify(message));
+};
+
+const errorBadRequest = (res: NextApiResponse, message: string) : void => {
+  writeResponse(res, 'application/problem+json', 400, {
+    title: 'bad request',
+    detail: message,
+  });
+};
+
+export const errorNotFoundRequest = (res: NextApiResponse) : void => {
+  writeResponse(res, 'application/json', 404, {
+    title: 'not found',
+  });
+};
+
+type Param = string | Number | Array<string | Number> | null;
+
+const queryParam = (req: NextApiRequest, key: string, defaultValue: Param = null) : Param => req.query[key] ?? defaultValue;
+
+export const reviewedPreprintSnippet = (manuscript: FullManuscriptConfig, meta?: MetaData) : ReviewedPreprintSnippet => {
+  const reviewed = reviewedDate(manuscript.status.timeline);
+
+  return {
+    id: manuscript.msid,
+    doi: manuscript.preprintDoi,
+    pdf: manuscript.pdfUrl,
+    status: 'reviewed',
+    authorLine: meta ? prepareAuthorLine(meta.authors) : undefined,
+    title: meta ? contentToHtml(meta.title) : undefined,
+    published: reviewed,
+    reviewedDate: reviewed,
+    versionDate: reviewed,
+    statusDate: reviewed,
+    stage: 'published',
+    subjects: getSubjects(manuscript.msas),
+  };
+};
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  const items = getManuscriptsLatest(config.manuscriptConfigFile).map((manuscript) => reviewedPreprintSnippet(manuscript));
+
+  const [perPage, page] = [
+    queryParam(req, 'per-page', 20),
+    queryParam(req, 'page', 1),
+  ].map((v) => {
+    const n = Number(v);
+
+    return n.toString() === parseInt(n.toString(), 10).toString() ? n : -1;
+  });
+  const order = queryParam(req, 'order', 'desc');
+
+  if (page <= 0) {
+    errorBadRequest(res, 'expecting positive integer for \'page\' parameter');
+  }
+
+  if (perPage <= 0 || perPage > 100) {
+    errorBadRequest(res, 'expecting positive integer between 1 and 100 for \'per-page\' parameter');
+  }
+
+  if (typeof order !== 'string' || ['asc', 'desc'].includes(order) === false) {
+    errorBadRequest(res, 'expecting either \'asc\' or \'desc\' for \'order\' parameter');
+  }
+
+  const offset = (page - 1) * perPage;
+
+  writeResponse(res, 'application/vnd.elife.reviewed-preprint-list+json; version=1', 200, {
+    total: items.length,
+    items: await Promise
+      .all(
+        (
+          items
+            // Order is not working when per-page is set.
```

Comment:
Could we pair on this one. I'm not really sure why this isn't working.

Reply:
👍